server/bin/pbench-backup-tarballs.py

```diff
@@ -0,0 +1,274 @@
+        except Exception:
+            _logger.exception(
+                "os.mkdir: Unable to create backup destination directory: {}".format(backup_controller_path))
+            raise
+        backup_contents = os.listdir(backup_controller_path)
+        archive_md5 = ("{}.md5".format(tar))
+        with open(archive_md5) as f:
+            archive_md5_hex_value = f.readline().split(" ")[0]
+
+        # Check if the result already present in backup or not
+        if resultname in backup_contents:
+            backup_md5 = ("{}/{}.md5".format(backup_controller_path, resultname))
+            with open(backup_md5) as f:
+                backup_md5_hex_value = f.readline().split(" ")[0]
+            if archive_md5_hex_value == backup_md5_hex_value:
+                # declare success
+                _logger.info(
+                    "Backed-up successfully: {}\n".format(resultname))
+            else:
+                # quarantine the result and move the symlink
+                # from _linksrc to _linkdest
+                quarantine(qdir, _logger, tar, archive_md5)
+                _rename_tb_link(tb, os.path.join(
+                    controller_path, _linkdest), _logger)
+                _logger.info(
+                    "Quarantine result since {} already exists in backup\n".format(resultname))
+                continue
+        else:
+            try:
+                shutil.copy(tar, backup_controller_path)
+                shutil.copy(archive_md5, backup_controller_path)
+                _logger.info(
+                    "Backed-up successfully: {}\n".format(resultname))
+            except Exception:
+                _logger.exception(
+                    "os.rename: Unable to move tar ball link {} and its md5 {}to destination directory: {}".format(tar, archive_md5, backup_dest))
+                raise
+
+        # Check if the result already present in s3 or not
+        if s3_resultname in s3_contents:
+            # move the symlink from _linksrc to _linkdest and
+            # declare success
+            _rename_tb_link(tb, os.path.join(
+                controller_path, _linkdest), _logger)
+            _logger.info(
+                "Upload to s3 successfully: {}\n".format(s3_resultname))
+        else:
+            md5_base64_value = (base64.b64encode(
+                bytes.fromhex(archive_md5_hex_value))).decode()
+            try:
+                with open(tar, 'rb') as data:
+                    s3_obj.connector.put_object(
+                        Bucket=s3_obj.bucket_name, Key=s3_resultname, Body=data, ContentMD5=md5_base64_value)
+            except Exception as e:
+                _logger.info(
+                    "Bad md5 for: {}, {}\n".format(s3_resultname, e))
+            _rename_tb_link(tb, os.path.join(
+                controller_path, _linkdest), _logger)
+            _logger.info(
+                "Upload to s3 successfully: {}\n".format(s3_resultname))
+
+
+def backup_data_to_s3(s3_obj, config, report):
```

Comment:
I think this has outlived its usefulness: in particular, the error messages are not accurate any longer.